cardano-db-sync/src/Cardano/DbSync/OffChain.hs

```diff
@@ -0,0 +1,280 @@
+  ReaderT SqlBackend m ()
+insertOffChainVoteResults trce resultQueue = do
+  res <- liftIO . atomically $ flushTBQueue resultQueue
+  unless (null res) $ do
+    let resLength = length res
+        resErrorsLength = length $ filter isFetchError res
+    liftIO . logInfo trce $
+      logInsertOffChainResults "Voting Anchor" resLength resErrorsLength
+  mapM_ insert res
+  where
+    insert :: (MonadBaseControl IO m, MonadIO m) => OffChainVoteResult -> ReaderT SqlBackend m ()
+    insert = \case
+      OffChainVoteResultMetadata md -> void $ DB.insertOffChainVoteData md
+      OffChainVoteResultError fe -> void $ DB.insertOffChainVoteFetchError fe
+
+    isFetchError :: OffChainVoteResult -> Bool
+    isFetchError = \case
+      OffChainVoteResultMetadata {} -> False
+      OffChainVoteResultError {} -> True
+
+logInsertOffChainResults ::
+  Text -> -- Pool of Vote
+  Int -> -- length of tbQueue
+  Int -> -- length of errors in tbQueue
+  Text
+logInsertOffChainResults offChainType resLength resErrorsLength =
+  mconcat
+    [ offChainType
+    , " Offchain "
+    , "metadata fetch: "
+    , DB.textShow (resLength - resErrorsLength)
+    , " results, "
+    , DB.textShow resErrorsLength
+    , " fetch errors"
+    ]
+
+---------------------------------------------------------------------------------------------------------------------------------
+-- Run OffChain threads
+---------------------------------------------------------------------------------------------------------------------------------
+runFetchOffChainPoolThread :: SyncEnv -> IO ()
+runFetchOffChainPoolThread syncEnv = do
+  -- if dissable gov is active then don't run voting anchor thread
+  when (ioOffChainPoolData iopts) $ do
+    logInfo trce "Running Offchain Pool fetch thread"
+    forever $ do
+      tDelay
+      -- load the offChain vote work queue using the db
+      _ <- runReaderT (loadOffChainPoolWorkQueue trce (envOffChainPoolWorkQueue syncEnv)) (envBackend syncEnv)
+      poolq <- blockingFlushTBQueue (envOffChainPoolWorkQueue syncEnv)
+      manager <- Http.newManager tlsManagerSettings
+      now <- liftIO Time.getPOSIXTime
+      mapM_ (queuePoolInsert <=< fetchOffChainPoolData trce manager now) poolq
+  where
+    trce = getTrace syncEnv
+    iopts = getInsertOptions syncEnv
+
+    queuePoolInsert :: OffChainPoolResult -> IO ()
+    queuePoolInsert = atomically . writeTBQueue (envOffChainPoolResultQueue syncEnv)
+
+runFetchOffChainVoteThread :: SyncEnv -> IO ()
+runFetchOffChainVoteThread syncEnv = do
+  -- if dissable gov is active then don't run voting anchor thread
+  when (ioGov iopts) $ do
+    logInfo trce "Running Offchain Vote Anchor fetch thread"
+    forever $ do
+      tDelay
+      -- load the offChain pool work queue using the db
+      _ <- runReaderT (loadOffChainVoteWorkQueue trce (envOffChainVoteWorkQueue syncEnv)) (envBackend syncEnv)
+      voteq <- blockingFlushTBQueue (envOffChainVoteWorkQueue syncEnv)
```

Comment:
I think this is what causes the inexistence of data in sanchonet. It loads then it flushes and blocks. If at any point there are no data to load then `blockingFlushTBQueue` will block indefinitely. This causes a dead end, the Queue (currently) is not filled by a separate thread, so there is no way to make progress. One solution is to use the non blocking variant of `blockingFlushTBQueue`